### PR TITLE
Audio DFU ADV name based on role

### PR DESCRIPTION
**Pull request by pirun**
1. Instead of advertising fix name NRF5340_AUDIO when in DFU mode, now ADV name will be combination of CONFIG_BT_DEVICE_NAME and role of gateway or headset.

### CODE REVIEW

File: applications/nrf5340_audio/README.rst [hunk 1]
```diff
@@ -861,6 +861,13 @@ Entering the DFU mode
 The |NCS| uses :ref:`SMP server and mcumgr <zephyr:device_mgmt>` as the DFU backend.
 Unlike the CIS and BIS modes for gateway and headsets, the DFU mode is advertising using the SMP server service.
 For this reason, to enter the DFU mode, you must long press **BTN 4** during each device startup to have the nRF5340 Audio DK enter the DFU mode.
+To distinguish the device role, DFU mode advertising name will be based on role and the pattern : ``CONFIG_BT_DEVICE_NAME`` _ **ROLE** _DFU.
+The name will look like:
+
+* BIS Gateway: Broadcast Audio Source_GW_DFU
```
Comment (rick1082) [inline applications/nrf5340_audio/README.rst:867]: We might need to sync up the ADV name between BIS and CIS for making it consistency.

Comment (alexsven) [reply applications/nrf5340_audio/README.rst:867]: Agree, I don't think it's necessary to differentiate between BIS and CIS. All should be NRF5340_AUDIO_...

File: applications/nrf5340_audio/src/modules/dfu_entry.c [hunk 2]
```diff
@@ -3,13 +3,17 @@
  *
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
+#define _BSD_SOURCE
```
Comment (alexsven) [inline applications/nrf5340_audio/src/modules/dfu_entry.c:6]: What is this, and why is it above the #include?

**Review by alexsven — CHANGES_REQUESTED** [review]

File: applications/nrf5340_audio/src/modules/dfu_entry.c (hunk 2 above)
Comment (pirun) [reply applications/nrf5340_audio/src/modules/dfu_entry.c:6]: I'm using strlcpy, strlcat -- size-bounded string copying and concatenation function here.
The prototype in zephyr-sdk-0.14.2\arm-zephyr-eabi\arm-zephyr-eabi\include\string.h is defined by pre-processor __BSD_VISIBLE.
__BSD_VISIBLE is defined by pre-processor in zephyr-sdk-0.14.2\arm-zephyr-eabi\arm-zephyr-eabi\include\sys\features.h.

The #define _BSD_SOURCE should be before the first instance of #include string.h. Mind that string.h could be included by some other #include statements in your source file, so the easiest is to put the #define _BSD_SOURCE on the very first line of the source file.

File: applications/nrf5340_audio/README.rst [hunk 3]
```diff
@@ -861,6 +861,13 @@ Entering the DFU mode
 The |NCS| uses :ref:`SMP server and mcumgr <zephyr:device_mgmt>` as the DFU backend.
 Unlike the CIS and BIS modes for gateway and headsets, the DFU mode is advertising using the SMP server service.
 For this reason, to enter the DFU mode, you must long press **BTN 4** during each device startup to have the nRF5340 Audio DK enter the DFU mode.
+To distinguish the device role, DFU mode advertising name will be based on role and the pattern : ``CONFIG_BT_DEVICE_NAME`` _ **ROLE** _DFU.
```
Comment (greg-fer) [inline applications/nrf5340_audio/README.rst:864]: Quite a few questions:
1. How does this information relate to entering the DFU?
2. Where is the name displayed?
3. What is the relation of the pattern against lines 867-870.
4. When is this Kconfig set? It doesn't appear anywhere in the doc except for here.

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: applications/nrf5340_audio/src/modules/dfu_entry.c (hunk 2 above)
Comment (erikrobstad) [inline applications/nrf5340_audio/src/modules/dfu_entry.c:6]: Should add a newline between the license and this #define.
Maybe also add a comment to the code why this define is needed?

File: applications/nrf5340_audio/src/modules/dfu_entry.c [hunk 4]
```diff
@@ -3,13 +3,17 @@
...
 #include <zephyr/kernel.h>
 #include <bluetooth/bluetooth.h>
 #include <bluetooth/conn.h>
+#include "string.h"
 #include "button_assignments.h"
 #include "ble_core.h"
 #include "button_handler.h"
 #include "macros_common.h"
+#include "channel_assignment.h"
+
```
Comment (erikrobstad) [inline applications/nrf5340_audio/src/modules/dfu_entry.c:16]: No need to add a newline here

File: applications/nrf5340_audio/src/modules/dfu_entry.c [hunk 5]
```diff
@@ -87,15 +91,40 @@ static struct bt_conn_cb dfu_conn_callbacks = {
...
 {
-	char name[CONFIG_BT_DEVICE_NAME_MAX];
+	char name[CONFIG_BT_DEVICE_NAME_MAX] = {0};
+
+	strlcpy(name, CONFIG_BT_DEVICE_NAME, CONFIG_BT_DEVICE_NAME_MAX);
+	strlcat(name, "_", CONFIG_BT_DEVICE_NAME_MAX);
+#if (CONFIG_AUDIO_DEV == GATEWAY)
+	strlcat(name, GW_TAG, CONFIG_BT_DEVICE_NAME_MAX);
+#else
+	int ret;
```
Comment (erikrobstad) [inline applications/nrf5340_audio/src/modules/dfu_entry.c:103]: Nitpick: These should be declared on the top of the function

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py [hunk 6]
```diff
@@ -39,17 +39,26 @@ def awklike(field_str, filename):
...
+    indices = [i for i, c in enumerate(build_dir) if c == folder_slash]
+    final_file_prefix = build_dir[indices[-2]+1:].replace(folder_slash, '_')+'_'
 
     # RETEIVE setting value from .config
     # "${ZEPHYR_BASE}/../bootloader/mcuboot/root-rsa-2048.pem"
 
     mcuboot_rsa_key = awklike('CONFIG_BOOT_SIGNATURE_KEY_FILE=', build_dir +
                               '/mcuboot/zephyr/.config')
-    if ('/' in mcuboot_rsa_key) or ('\\' in mcuboot_rsa_key):
+    # Use \\ or / should be worked on OS windows
```
Comment (erikrobstad) [inline applications/nrf5340_audio/tools/buildprog/pcft_sign.py:57]: Change comment to something like this: "'\' is used for Windows, '\\' is used for other OSs like Linux'"

File: doc/nrf/releases/release-notes-changelog.rst [hunk 7]
```diff
@@ -176,6 +176,7 @@ nRF5340 Audio
   * Bonding between gateway and headsets in the CIS (Connected Isochronous Stream).
   * DFU support for internal and external flash layouts.
     See :ref:`nrf53_audio_app_configuration_configure_fota` in the application documentation for details.
+  * DFU advertising name base on role.
```
Comment (erikrobstad) [inline doc/nrf/releases/release-notes-changelog.rst:179]: base = based

**Review by erikrobstad — CHANGES_REQUESTED** [review]

File: applications/nrf5340_audio/README.rst (hunk 1 above)
Comment (pirun) [reply applications/nrf5340_audio/README.rst:867]: Are we going to make all BT_DEVICE_NAME the same in this PR?
Like remove [here](https://github.com/nrfconnect/sdk-nrf/blob/main/applications/nrf5340_audio/src/bluetooth/Kconfig.defaults#L197) and [here](https://github.com/nrfconnect/sdk-nrf/blob/main/applications/nrf5340_audio/src/bluetooth/Kconfig.defaults#L230).

File: applications/nrf5340_audio/src/modules/dfu_entry.c (hunk 5 above)
Comment (pirun) [reply applications/nrf5340_audio/src/modules/dfu_entry.c:103]: If CONFIG_AUDIO_DEV is GATEWAY, ret and channel will become unused variables.

File: applications/nrf5340_audio/src/modules/dfu_entry.c (hunk 2 above)
Comment (pirun) [reply applications/nrf5340_audio/src/modules/dfu_entry.c:8]: Add explanation here.

File: applications/nrf5340_audio/src/modules/dfu_entry.c (hunk 4 above)
Comment (pirun) [reply applications/nrf5340_audio/src/modules/dfu_entry.c:18]: Remove newline.

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py (hunk 6 above)
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/pcft_sign.py]: Change comment to more clear.

File: doc/nrf/releases/release-notes-changelog.rst (hunk 7 above)
Comment (pirun) [reply doc/nrf/releases/release-notes-changelog.rst]: Typo fixed.

File: applications/nrf5340_audio/src/modules/dfu_entry.c (hunk 2 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/modules/dfu_entry.c:8]: I can not see any explanation here in the newest version of the code?

File: applications/nrf5340_audio/README.rst (hunk 1 above)
Comment (rick1082) [reply applications/nrf5340_audio/README.rst:867]: I suggest that all the headset in DFU mode sharing the same name "NRF5340_AUDIO_GW_DFU"
And all the gateway in DFU mode sharing the same name "NRF5340_AUDIO_XX_DFU"

File: applications/nrf5340_audio/README.rst (hunk 3 above)
Comment (pirun) [reply applications/nrf5340_audio/README.rst:864]: We decide to unify BT_DEVICE_NAME for CIS and BIS.

**Review by alexsven — APPROVED** [review]

File: applications/nrf5340_audio/README.rst (hunk 3 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/README.rst:864]: Maybe we can change "CONFIG_BT_DEVICE_NAME" to just "NRF5340_AUDIO"?
(Meaning that we do not explicitly say in this doc that the name comes from this config)

File: applications/nrf5340_audio/README.rst [hunk 8]
```diff
@@ -861,6 +861,12 @@ Entering the DFU mode
...
+The name will look like:
+
+* Gateway: NRF5340_AUDIO_GW_DFU
+* Headset: NRF5340_AUDIO_HL_DFU
```
Comment (erikrobstad) [inline applications/nrf5340_audio/README.rst:868]: I suggest changing to something like:
* Left headset: (...)
* Right headset: (...)

File: applications/nrf5340_audio/src/bluetooth/Kconfig.defaults [hunk 9]
```diff
@@ -27,12 +27,11 @@ config BT_EXT_ADV
...
 	default y
 
-if TRANSPORT_CIS # Depends on TRANSPORT_CIS
-
 config BT_DEVICE_NAME
 	string
 	default "NRF5340_AUDIO"
 
+if TRANSPORT_CIS # Depends on TRANSPORT_CIS
 config BT_GATT_CLIENT
```
Comment (erikrobstad) [inline applications/nrf5340_audio/src/bluetooth/Kconfig.defaults:35]: Add newline here

**Review by erikrobstad — CHANGES_REQUESTED** [review]

File: applications/nrf5340_audio/README.rst (hunk 3 above)
Comment (pirun) [reply applications/nrf5340_audio/README.rst:864]: CONFIG_BT_DEVICE_NAME is configurable. Customer might change the value from "NRF5340_AUDIO" to another value.
How about add a line * Default ``CONFIG_BT_DEVICE_NAME``: NRF5340_AUDIO. Then the name pattern.

File: applications/nrf5340_audio/README.rst (hunk 1 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/README.rst]: Resolving this conversation. We have agreed that we want to make all device types have the same BT_DEVICE_NAME in this PR.

File: applications/nrf5340_audio/README.rst [hunk 10]
```diff
@@ -847,6 +847,7 @@ See the following table for the pin definitions.
 +-------------+-------------------+-------------+
 | P1.10       | CS                | D8          |
 +-------------+-------------------+-------------+
+That external flash shield must be connected for the kits to boot, even if DFU mode is not initiated.
```
Comment (erikrobstad) [inline applications/nrf5340_audio/README.rst:850]:
```suggestion
.. note::
   External flash shields must be connected for the kits to boot, even if DFU mode is not 
   initiated.
```

File: applications/nrf5340_audio/README.rst (hunk 3 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/README.rst:865]:
```suggestion
To distinguish the device types, DFU mode advertising names will be based on these. The advertising names follow this pattern: NRF5340_AUDIO_ **ROLE** _DFU. The first part of the names is based on ``CONFIG_BT_DEVICE_NAME`` . 
```

**Review by erikrobstad — APPROVED** [review]

File: applications/nrf5340_audio/README.rst [hunk 11]
```diff
@@ -861,6 +864,12 @@ Entering the DFU mode
 The |NCS| uses :ref:`SMP server and mcumgr <zephyr:device_mgmt>` as the DFU backend.
 Unlike the CIS and BIS modes for gateway and headsets, the DFU mode is advertising using the SMP server service.
 For this reason, to enter the DFU mode, you must long press **BTN 4** during each device startup to have the nRF5340 Audio DK enter the DFU mode.
+To distinguish the device types, DFU mode advertising names will be based on these. The advertising names follow this pattern: NRF5340_AUDIO\_ **ROLE** _DFU.
+The first part of the names is based on ``CONFIG_BT_DEVICE_NAME`` .
+
+* Gateway: NRF5340_AUDIO_GW_DFU
+* Left Headset: NRF5340_AUDIO_HL_DFU
+* Right Headset: NRF5340_AUDIO_HR_DFU
```
Comment (greg-fer) [inline applications/nrf5340_audio/README.rst:872]:
```suggestion

To identify the devices before the DFU takes place, the DFU mode advertising names mention the device type directly.
The names follow the pattern in which the device *ROLE* is inserted before the ``_DFU`` suffix.
For example:

* Gateway: NRF5340_AUDIO_GW_DFU
* Left Headset: NRF5340_AUDIO_HL_DFU
* Right Headset: NRF5340_AUDIO_HR_DFU

The first part of these names is based on :kconfig:option:`CONFIG_BT_DEVICE_NAME`.
```
Not sure what is meant with "the first part of the names" in the original...

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: applications/nrf5340_audio/README.rst (hunk 11 above)
Comment (pirun) [reply applications/nrf5340_audio/README.rst:872]: The first part of the names is "NRF5340_AUDIO".

File: applications/nrf5340_audio/README.rst [hunk 12]
```diff
@@ -862,6 +865,16 @@ The |NCS| uses :ref:`SMP server and mcumgr <zephyr:device_mgmt>` as the DFU back
...
 
+To identify the devices before the DFU takes place, the DFU mode advertising names mention the device type directly.
+The names follow the pattern in which the device *ROLE* is inserted before the ``_DFU`` suffix.
+For example:
+
+* Gateway: NRF5340_AUDIO_GW_DFU
+* Left Headset: NRF5340_AUDIO_HL_DFU
+* Right Headset: NRF5340_AUDIO_HR_DFU
+
+The first part of these names is based on :kconfig:option:`CONFIG_BT_DEVICE_NAME`
```
Comment (greg-fer) [inline applications/nrf5340_audio/README.rst:876]:
```suggestion
The first part of these names is based on :kconfig:option:`CONFIG_BT_DEVICE_NAME`.
```

**Review by greg-fer — APPROVED** [review]
Approving, but there is a period missing. It's important, even if it is just a period. :D Kindly please add it :)

**Pull request merged**